### PR TITLE
Implement the rest of the router interface

### PR DESCRIPTION
**Pull request by sourabhmarathe**

### CODE REVIEW

File: build/marketplace/Interfaces.sol [hunk 1]
```diff
@@ -69,4 +71,50 @@ interface IPool {
     function sellPrincipalTokenPreview(uint128) external returns (uint128);
 
     function buyPrincipalTokenPreview(uint128) external returns (uint128);
+
+    function mint(
```
Comment (JTraversa) [inline build/marketplace/Interfaces.sol]: format all these into 1 line (just how we do our interfaces)

File: build/marketplace/MarketPlace.sol [hunk 2]
```diff
@@ -49,6 +49,25 @@ contract MarketPlace {
         redeemer = r;
     }
 
+    /// @notice allows the admin to create a market
+    /// @param p enum value of the principal token
+    /// @param u underlying token address
+    /// @param m maturity timestamp for the market
+    /// @param a address of the new market
+    /// @return bool true if successful
+    function setMarket(
```
Comment (JTraversa) [inline build/marketplace/MarketPlace.sol]: setPrincipal instead of setMarket

File: build/marketplace/MarketPlace.sol [hunk 3]
```diff
@@ -188,11 +205,111 @@ contract MarketPlace {
...
+    )
+        external
+        returns (
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        IPool pool = IPool(pools[u][m]);
+        address hello = pool.underlying();
```
Comment (JTraversa) [inline build/marketplace/MarketPlace.sol]: hello there

?

File: build/marketplace/MarketPlace.sol [hunk 4]
```diff
@@ -188,11 +205,111 @@ contract MarketPlace {
...
+    /// @return The amount of liquidity tokens minted.
+    function mint(
+        address u,
+        uint256 m,
+        uint256 underlyingAmount,
+        uint256 principalTokenAmount,
+        uint256 minRatio,
+        uint256 maxRatio
+    )
+        external
```
Comment (JTraversa) [inline build/marketplace/MarketPlace.sol]: single line for the external, modifiers, and returns (just keeping format with our other methods)

File: build/marketplace/MarketPlace.sol (hunk 3 above)
Comment (sourabhmarathe) [reply build/marketplace/MarketPlace.sol]: oops, was testing, will revert lol

**Pull request merged**